darts/timeseries.py

```diff
@@ -929,8 +957,21 @@ def _combine_from_pd_ops(self, other: 'TimeSeries',
         raise_if_not(self.has_same_time_as(other), 'The two TimeSeries must have the same time index.', logger)
         raise_if_not(self._df.shape == other._df.shape, 'The two TimeSeries must have the same shape.', logger)
 
-        series = combine_fn(self._df, other.pd_dataframe())
-        return TimeSeries(series, self.freq())
+        df_a = self._df
+        df_b = other._df
+        if df_a is not None and df_b is not None:
+            # univariate case
+            if df_a.shape[1] == 1:
+                pd_serie_a = df_a.iloc[:, 0]
+                pd_serie_b = df_b.iloc[:, 0]
+                pd_serie = combine_fn(pd_serie_a, pd_serie_b)
+                series_df = pd_serie.to_frame()
+            # multivariate case
+            else:
+                raise_if(len(set(df_a.columns.to_list() + df_b.columns.columns.to_list())) != len(df_a.columns),
+                         "Column name in each TimeSeries must match one to one.")
+                series_df = combine_fn(df_a, df_b)
+        return TimeSeries(series_df, self.freq(), columns=series_df.columns)
```

Comment:
If both `df_a` and `df_b` are None that will throw a not nice error saying `series_df` doesn't exist.

Reply:
In addition, to respect types, you should pass `self.freq_str()` instead of `self.freq()`. I even wonder if that might not lead to errors, and now that I speak of it, I see we have this issue at several places in the code.

Reply:
1) you are right and its even worst because it is also the case when either one or the other is None

2) I used the syntax previously here but you are right I can fix that as well

Reply:
Good point about the `freq` parameter. I actually started this trend of passing `self.freq()`, sorry!

Reply:
1. I just removed the condition since already from type they should not be None. The error will then be when calling .shape if they happens to be None

2. Fixed it